### PR TITLE
fix #15628

### LINKED ISSUE #15628: coro lib doesn't work with orc/arc
Coroutines doesn't compile with gc orc/arc

coro.nim uses nimGC_setStackBottom and references GC_addStack, etc. implemented in gc_common.nim which is not included by orc / arc

as of 
Nim Compiler Version 1.5.1 [Windows: amd64]
git hash: 091ec916fd2737d495c5ef54afb5964ee62c049d

### PR DESCRIPTION
**Pull request by ringabout**
continue #15811

### CODE REVIEW

**Comment by timotheecour** [comment]
CI fails on openbsd:
> CC: stdlib_coro.nim
/home/build/Nim/nimcache/tests/coroutines/tgc.nim_4a8a08f09d37b73795649038408b5f33/stdlib_coro.nim.c:393:22: error: array type 'jmp_buf' (aka 'long [11]') is not assignable
        (*dest).execContext = src.execContext;

openbsd could be disabled to unblock this PR (assuming the part that'd be disabled would be already not working prior to this PR); however, I wonder whether this is the right approach or whether `nimGC_setStackBottom` + friends should be defined for arc/orc?

**Comment by Araq** [comment]
> whether nimGC_setStackBottom + friends should be defined for arc/orc?

They should not.

File: tests/coroutines/tgc.nim [hunk 1]
```diff
@@ -1,5 +1,6 @@
 discard """
-  targets: "c"
+  matrix: "--gc:refc; --gc:arc; --gc:orc"
+  target: "c"
```
Comment (timotheecour) [inline tests/coroutines/tgc.nim:3]:
```suggestion
  target: "c"
  disabled: "openbsd"
```

File: tests/coroutines/twait.nim [hunk 2]
```diff
@@ -1,6 +1,7 @@
 discard """
   output: "Exit 1\nExit 2"
-  targets: "c"
+  matrix: "--gc:refc; --gc:arc; --gc:orc"
+  target: "c"
```
Comment (timotheecour) [inline tests/coroutines/twait.nim:4]:
```suggestion
  target: "c"
  disabled: "openbsd"
```

File: tests/coroutines/tgc.nim [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
 """
```
Comment (timotheecour) [inline tests/coroutines/tgc.nim:5]:
```suggestion
# openbsd gave: stdlib_coro.nim.c:406:22: error: array type 'jmp_buf' (aka 'long [11]') is not assignable (*dest).execContext = src.execContext;
```

**Review by timotheecour — APPROVED** [review]

**Pull request merged**